pennylane/tape/qnode.py

```diff
@@ -323,7 +346,41 @@ def _validate_reversible_method(device, interface):
                 f"The {device.short_name} device does not support reversible differentiation."
             )
 
-        return ReversibleTape, interface, "analytic", device
+        return ReversibleTape, interface, device, {"method": "analytic"}
+
+    @staticmethod
+    def _validate_adjoint_method(device, interface):
+        """Validates whether a particular device and JacobianTape interface
+        supports the ``"adjoint"`` differentiation method.
+
+        Args:
+            device (.Device): PennyLane device
+            interface (str): name of the requested interface
+
+        Returns:
+            tuple[.JacobianTape, str, .Device, dict[str, str]]: Tuple containing the compatible
+            JacobianTape, the interface to apply, the device to use, and the method argument
+            to pass to the ``JacobianTape.jacobian`` method.
+
+        Raises:
+            qml.QuantumFunctionError: if the device does not support adjoint backprop
+        """
+        supported_device = hasattr(device, "_apply_operation")
+        supported_device = supported_device and hasattr(device, "_apply_unitary")
+        supported_device = supported_device and device.capabilities().get("returns_state")
+        supported_device = supported_device and hasattr(device, "adjoint_jacobian")

```

Comment:
Are these the only and all requirements for the adjoint method to work?

Reply:
Probably a reasonably good approximation :laughing:

For example, we require `_apply_operation` and `_apply_unitary()` to *return* the state (which is not checked above). We also require the state to be pure (the presence of `_apply_unitary` above is what excludes `default.mixed`).

Finally, we are currently requiring the circuit to only have `expval` measurements. We can't check that at this point, so the check is deferred to `adjoint_jacobian()`

Reply:
It would be great to come back to the device requirements and make them a bit more rigorous.

Reply:
Ok, thanks @trbromley. I think it makes sense having it like this for now, but, as you said, it might be nice to make this a bit more rigorous in the future. This might warrant a comment/comments stating why these requirements are here and that they might not be bullet proof.

Reply:
Good points! Have added a note.